pkg/volume/csi/csi_attacher.go

```diff
@@ -287,7 +287,9 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 	if c.csiClient == nil {
 		c.csiClient, err = newCsiDriverClient(csiDriverName(csiSource.Driver))
 		if err != nil {
-			return errors.New(log("attacher.MountDevice failed to create newCsiDriverClient: %v", err))
+			// Treat the absence of the CSI driver as a transient error
+			// See https://github.com/kubernetes/kubernetes/issues/120268
+			return volumetypes.NewTransientOperationFailure(log("attacher.MountDevice failed to create newCsiDriverClient: %v", err))
```

Comment:
Doing a quick scan of other places where we should return transient error:
UnstageVolume: https://github.com/kubernetes/kubernetes/blob/689285c4101a2035a012cd275b8d1996d71ba74d/pkg/volume/csi/csi_attacher.go#L612

Teardown: https://github.com/kubernetes/kubernetes/blob/689285c4101a2035a012cd275b8d1996d71ba74d/pkg/volume/csi/csi_mounter.go#L422

SetUpDevice:
https://github.com/kubernetes/kubernetes/blob/689285c4101a2035a012cd275b8d1996d71ba74d/pkg/volume/csi/csi_block.go#L322

MapDevice: https://github.com/kubernetes/kubernetes/blob/689285c4101a2035a012cd275b8d1996d71ba74d/pkg/volume/csi/csi_block.go#L382

TeardownDevice: https://github.com/kubernetes/kubernetes/blob/689285c4101a2035a012cd275b8d1996d71ba74d/pkg/volume/csi/csi_block.go#L447

UnmapDevice: https://github.com/kubernetes/kubernetes/blob/689285c4101a2035a012cd275b8d1996d71ba74d/pkg/volume/csi/csi_block.go#L509

NodeExpand? https://github.com/kubernetes/kubernetes/blob/689285c4101a2035a012cd275b8d1996d71ba74d/pkg/volume/csi/expander.go#L49

@gnufied @jsafrane wdyt?

Reply:
@msau42: I'll upload a patch to treat all of these errors as transient too.